store/sqlstore/session_store.go

```diff
@@ -45,37 +45,31 @@ func (me SqlSessionStore) CreateIndexesIfNotExists() {
 	me.CreateIndexIfNotExists("idx_sessions_last_activity_at", "Sessions", "LastActivityAt")
 }
 
-func (me SqlSessionStore) Save(session *model.Session) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if len(session.Id) > 0 {
-			result.Err = model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.existing.app_error", nil, "id="+session.Id, http.StatusBadRequest)
-			return
-		}
+func (me SqlSessionStore) Save(session *model.Session) (*model.Session, *model.AppError) {
+	if len(session.Id) > 0 {
+		return nil, model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.existing.app_error", nil, "id="+session.Id, http.StatusBadRequest)
+	}
 
-		session.PreSave()
+	session.PreSave()
 
-		tcs := me.Team().GetTeamsForUser(session.UserId)
+	tcs := me.Team().GetTeamsForUser(session.UserId)
 
-		if err := me.GetMaster().Insert(session); err != nil {
-			result.Err = model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.app_error", nil, "id="+session.Id+", "+err.Error(), http.StatusInternalServerError)
-			return
-		} else {
-			result.Data = session
-		}
+	if err := me.GetMaster().Insert(session); err != nil {
+		return nil, model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.app_error", nil, "id="+session.Id+", "+err.Error(), http.StatusInternalServerError)
+	}
 
-		if rtcs := <-tcs; rtcs.Err != nil {
-			result.Err = model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.app_error", nil, "id="+session.Id+", "+rtcs.Err.Error(), http.StatusInternalServerError)
-			return
-		} else {
-			tempMembers := rtcs.Data.([]*model.TeamMember)
-			session.TeamMembers = make([]*model.TeamMember, 0, len(tempMembers))
-			for _, tm := range tempMembers {
-				if tm.DeleteAt == 0 {
-					session.TeamMembers = append(session.TeamMembers, tm)
-				}
+	if rtcs := <-tcs; rtcs.Err != nil {
```

Comment:
If you change this to something like:
```go
rtcs := <-tcs
if rtcs.Err != nil {
...
```
You can remove the else clause and unindent the entire block in to the "happy path". Which is more idiomatic about error handling.